src/domain/exchange/exchange.repository.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpException } from '@nestjs/common';
+import { JsonSchemaService } from '../../common/schemas/json-schema.service';
+import { ValidationErrorFactory } from '../errors/validation-error-factory';
+import { IExchangeApi } from '../interfaces/exchange-api.interface';
+import { ExchangeRepository } from './exchange.repository';
+import { ExchangeApi } from '../../datasources/exchange-api/exchange-api.service';
+import exchangeResultFactory from '../../domain/exchange/entities/__tests__/exchange.factory';
+import exchangeFiatCodesFactory from '../../domain/exchange/entities/__tests__/fiat-codes.factory';
+
+const validationErrorFactory = {
+  from: jest.fn().mockReturnValue(new HttpException('testErr', 500)),
```

Comment:
I'd say to mock any return value in the tests themselves. So the setup for the test is in one place.